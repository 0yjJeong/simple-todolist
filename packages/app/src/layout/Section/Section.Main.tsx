import styled from 'styled-components';

const Section = styled.div`
  text-align: center;
  margin-top: 100px;
  margin-bottom: 100px;

  a {
    margin-top: 40px;
    display: inline-block;
  }

  .typo-title {
    margin-bottom: 10px;
  }

  .typo-subTitle {
    color: #787878;
  }
`;

export default Section;
