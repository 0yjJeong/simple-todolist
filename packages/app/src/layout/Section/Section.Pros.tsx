import styled from 'styled-components';

const Section = styled.div`
  position: relative;
  text-align: center;
  margin-top: 6rem;
  margin-bottom: 6rem;

  &:before {
    content: '';
    display: block;
    height: 205px;
    background: #dbffe5;
    border-radius: 10px;
    position: absolute;
    width: 100%;
    z-index: -1;
  }

  & > * {
    margin-left: 1.4rem;
    margin-right: 1.4rem;
  }

  & > p {
    margin-top: 5rem;
    margin-bottom: 1rem;
  }
`;

export default Section;
