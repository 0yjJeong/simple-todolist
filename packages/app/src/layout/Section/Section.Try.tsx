import styled from 'styled-components';
import media from '../../styles/media';

const Section = styled.div`
  max-width: 460px;
  margin-left: auto;
  margin-right: auto;
  margin-top: 240px;
  margin-bottom: 6rem;
  text-align: center;

  ${media.modile} {
    width: 100%;
  }
`;

export default Section;
