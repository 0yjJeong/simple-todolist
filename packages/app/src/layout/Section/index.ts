const Section = {
  Main: require('./Section.Main').default,
  Cards: require('./Section.Cards').default,
  Try: require('./Section.Try').default,
};

export default Section;
