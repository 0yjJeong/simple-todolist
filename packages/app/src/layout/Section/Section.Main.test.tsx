import { render } from '@testing-library/react';
import 'jest-styled-components';
import Section from './Section.Main';

describe('<Section.Main/>', () => {
  it('should works', () => {
    const rendered = render(<Section />);
    expect(rendered.container.firstChild).toMatchSnapshot();
    expect(rendered.container.firstChild).toHaveStyleRule(
      'text-align',
      'center'
    );
    expect(rendered.container.firstChild).toHaveStyleRule(
      'display',
      'inline-block',
      {
        modifier: 'a',
      }
    );
    expect(rendered.container.firstChild).toHaveStyleRule(
      'margin-bottom',
      '10px',
      {
        modifier: '.typo-title',
      }
    );
    expect(rendered.container.firstChild).toHaveStyleRule('color', '#787878', {
      modifier: '.typo-subTitle',
    });
  });
});
