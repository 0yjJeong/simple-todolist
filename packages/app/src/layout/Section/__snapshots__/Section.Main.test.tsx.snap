// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section.Main/> should works 1`] = `
.c0 {
  text-align: center;
  margin-top: 100px;
  margin-bottom: 100px;
}

<div
  class="c0"
/>
`;
